src/ring_pedersen.rs

```diff
@@ -0,0 +1,318 @@
+        let randomness = random_plusminus_scaled(rng, range, &(modulus * &self.modulus));
+        CommitmentRandomness(randomness)
+    }
+
+    /// Produces a commitment to `value`, which must be drawn
+    /// from `+/- 2^range`. The commitment randomness is sampled from
+    /// `+/- 2^range * N`, where `N` is the modulus of the commitment scheme.
+    pub(crate) fn commit(
+        &self,
+        value: &BigNumber,
+        range: usize,
+        rng: &mut (impl RngCore + CryptoRng),
+    ) -> (Commitment, CommitmentRandomness) {
+        let randomness = self.commitment_randomness(range, &BigNumber::one(), rng);
+        let com = self.reconstruct(value, randomness.as_masked());
+        (com, randomness)
+    }
+
+    /// Reconstructs a commitment to `value` using [`MaskedRandomness`] `randomness`.
+    /// The commitment is computed as in [`RingPedersen::commit`], except `randomness` is used
+    /// instead of it being generated by the commitment process.
+    pub(crate) fn reconstruct(
+        &self,
+        value: &BigNumber,
+        randomness: &MaskedRandomness,
+    ) -> Commitment {
+        let a = modpow(&self.s, value, &self.modulus);
+        let b = modpow(&self.t, &randomness.0, &self.modulus);
+        Commitment(a.modmul(&b, &self.modulus))
+    }
+
+    /// Combines two commitments with exponent `e` by computing `com0 * com1^e mod N`,
+    /// where `N` is the modulus of the commitment scheme.
+    pub(crate) fn combine(
+        &self,
+        com0: &Commitment,
+        com1: &Commitment,
+        e: &BigNumber,
+    ) -> Commitment {
+        Commitment(
+            com0.0
+                .modmul(&modpow(&com1.0, e, &self.modulus), &self.modulus),
+        )
+    }
+
+    /// Produces a commitment to `value` using [`Commitment`] `com`.
+    ///
+    /// In more detail, the following is computed: `com^value t^r mod N` where `r` falls
+    /// in the range `+/- 2^range * modulus * N`, where `N` is the modulus of the
+    /// commitment scheme.
+    pub(crate) fn commit_with_commitment(
+        &self,
+        com: &Commitment,
+        value: &BigNumber,
+        range: usize,
+        modulus: &BigNumber,
+        rng: &mut (impl RngCore + CryptoRng),
+    ) -> (Commitment, CommitmentRandomness) {
+        let randomness = self.commitment_randomness(range, modulus, rng);
+        let com = self.reconstruct_with_commitment(com, value, randomness.as_masked());
+        (com, randomness)
+    }
+
+    /// Reconstructs a commitment to `value` using [`Commitment`] `com` and [`MaskedRandomness`]
+    /// `randomness`.
+    pub(crate) fn reconstruct_with_commitment(
+        &self,
+        com: &Commitment,
+        value: &BigNumber,
+        randomness: &MaskedRandomness,
+    ) -> Commitment {
+        Commitment(modpow(&com.0, value, &self.modulus).modmul(
+            &modpow(&self.t, &randomness.0, &self.modulus),
+            &self.modulus,
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
nit: this is a personal pet peeve, but I prefer to name Rust tests without the word `test` in them (since they're run with `cargo test` and are labeled with `#[test]`, we can basically always tell that they're tests). 

I like names that describe what's being checked, like
- `verified_ring_pedersen_generation_works`
- `parameters_must_match_proof_to_verify`
etc.